### PR TITLE
Clearly state and cite cutting edge optimization approaches to determine phase angles

### PR DESCRIPTION
**Pull request by Jaybsoni**
**Summary:** Upon review, we realized that one of the cited approaches to compute the phase angles belongs in the "optimization" section. It's an important result and the cutting edge in methods for computing phase angles. For this reason we update the demo to add more emphasis to it.

### CODE REVIEW

**Review by KetpuntoG — APPROVED** [review]
LGTM :)

File: demonstrations/tutorial_apply_qsvt.py [hunk 1]
```diff
@@ -271,9 +271,9 @@ def loss_func(phi):
 plt.show()
 
 ###############################################################################
-# Awesome, we successfully optimized the phase angles! While we used a standard loss function
-# and optimizer, users have the freedom to explore any optimizer, loss function, and sampling
-# scheme when training the phase angles for QSVT.
+# Awesome, we successfully optimized the phase angles! While we used a simple loss function
+# and optimizer, more sophisticated optimization schemes have been presented in literature to
+# robustly train the phase angles for QSVT (see [#phaseeval]_).
```
Comment (soranjh) [inline demonstrations/tutorial_apply_qsvt.py:276]: "train" or "obtain"?

Comment (Jaybsoni) [reply demonstrations/tutorial_apply_qsvt.py:276]: Train 👍🏼

**Review by soranjh — APPROVED** [review]

**Comment by KetpuntoG** [comment]
@Jaybsoni ready to merge?

**Pull request merged**